<launch>
    <arg name="input_topic" default="/lidar_aeb/raw_points" />
    <!--arg name="no_ground_point_topic" default="/points_no_ground"/>
    <arg name="ground_point_topic" default="/points_ground"/-->
    <arg name="no_ground_point_topic" default="/filtered_points_no_ground"/>
    <arg name="ground_point_topic" default="/filtered_points_ground"/>
    <arg name="all_points_topic" default="/all_points"/>

    <!-- clip the points above sensor_height+clip_height -->
    <arg name="max_window_size" default="6.0"/>
    <!-- the height of lidar position  -->
    <!--<arg name="sensor_height" default="1.77"/>-->
    <arg name="max_distance" default="5.0"/>
    <!-- the min distance to be remove -->
    <!--<arg name="min_distance" default="2.0"/>-->
    <arg name="init_distance" default="0.5"/>
    <!-- the max distance to be remove -->
    <!--<arg name="max_distance" default="75.0"/>-->
    <arg name="cell_size" default="0.8"/>
    <!-- we use 32 lidar -->
    <!--<arg name="sensor_model" default="32"/>-->
    <arg name="slope" default="0.1"/>
    <!-- Num of Iteration -->
    <arg name="base" default="2.0"/>
    <!-- Num of LPR -->
    <arg name="negative" default="true"/>
    <!-- Seeds Threshold -->
    <arg name="extract_ground" default="true"/>


    <node pkg="pmf_ground_filter" type="pmf_ground_filter_node" name="pmf_ground_filter_node" output="screen">
        <param name="input_topic" value="$(arg input_topic)"/>
        <param name="no_ground_point_topic" value="$(arg no_ground_point_topic)"/>
        <param name="ground_point_topic" value="$(arg ground_point_topic)"/>
        <param name="all_points_topic" value="$(arg all_points_topic)"/>

        <param name="max_window_size" value="$(arg max_window_size)" />
        <param name="max_distance" value="$(arg max_distance)" />
        <param name="init_distance" value="$(arg init_distance)" />
        <param name="cell_size" value="$(arg cell_size)" />
        <param name="slope" value="$(arg slope)" />
        <param name="base" value="$(arg base)" />
        <param name="negative" value="$(arg negative)" />
        <param name="extract_ground" value="$(arg extract_ground)" />
        
    </node>
    
</launch>
